src/main/java/com/linkedin/xinfra/monitor/services/MultiClusterTopicManagementService.java

```diff
@@ -326,32 +338,37 @@ void maybeAddPartitions(int minPartitionNum) throws ExecutionException, Interrup
       Map<String, KafkaFuture<TopicDescription>> kafkaFutureMap =
           _adminClient.describeTopics(Collections.singleton(_topic)).values();
       KafkaFuture<TopicDescription> topicDescriptions = kafkaFutureMap.get(_topic);
-      List<TopicPartitionInfo> partitions = topicDescriptions.get().partitions();
-
-      int partitionNum = partitions.size();
-      if (partitionNum < minPartitionNum) {
-        LOGGER.info("{} will increase partition of the topic {} in the cluster from {}" + " to {}.",
-            this.getClass().toString(), _topic, partitionNum, minPartitionNum);
-        Set<Integer> blackListedBrokers = _topicFactory.getBlackListedBrokers(_adminClient);
-        Set<BrokerMetadata> brokers = new HashSet<>();
-        for (Node broker : _adminClient.describeCluster().nodes().get()) {
-          BrokerMetadata brokerMetadata = new BrokerMetadata(broker.id(), null);
-          brokers.add(brokerMetadata);
-        }
+      List<TopicPartitionInfo> partitions = topicDescriptions.get(_requestTimeout.toMillis(), TimeUnit.MILLISECONDS).partitions();
 
-        if (!blackListedBrokers.isEmpty()) {
-          brokers.removeIf(broker -> blackListedBrokers.contains(broker.id()));
-        }
+      final int currPartitionNum = partitions.size();
+      if (currPartitionNum >= requiredMinPartitionNum) {
+        LOGGER.info("{} will not increase partition of the topic {} in the cluster. Current partition count {} and '" +
```

Comment:
is tthis expected tto be the common case? if so - shuold this be  info?

Reply:
It is a common case. However, to troubleshoot the issue in which the KMF did not increase the partition number as expected, this `info` log line will be helpful since it provides the exact reason why KMF decided not to take any action to increase the partition number.

Reply:
+1 on Radai's comment, since it is the common case, the info log may flush the log from 10+ monitored clusters periodically.

Reply:
OK, I changed it to the `debug` logging level